sparse/server.go

```diff
@@ -1,18 +1,83 @@
 package sparse
 
 import (
-	"crypto/sha1"
-	"encoding/binary"
+	"bytes"
 	"encoding/gob"
-	"hash"
+	"fmt"
 	"net"
 	"os"
+	"sort"
 	"strconv"
 	"time"
 
-	"github.com/rancher/sparse-tools/log"
+	log "github.com/Sirupsen/logrus"
 )
 
+const verboseServer = true
+
+type requestCode int
+
+const (
+	requestMagic    requestCode = 31415926
+	syncRequestCode requestCode = 1 + iota
+	syncHole
+	syncData
+	syncDone
+)
+
+func (reqCode requestCode) String() string {
+	var s string
+	if reqCode&syncRequestCode == syncRequestCode {
```

Comment:
Good catch! I was copying online code using bitwise shifting enum definition and String() function, but then I thought there is no need to do bitwise shifting, just incrementing is good enough. So I modified that definition part but forgot the String() function. I bet debugging message has been wrong afterwards that I did not pay attention when tests were passing. 
Fixing it now.